environs/interface.go

```diff
@@ -436,5 +437,12 @@ type UpgradeStep interface {
 	Description() string
 
 	// Run executes the upgrade business logic.
-	Run() error
+	Run(UpgradeStepParams) error
```

Comment:
I did that because I was going to use the steps to determine the current best version. Then I added EnvironProvider.Version() instead. I'll revert this change.